src/Protocol/Document/TextDocumentAttributes.cs

```diff
@@ -2,8 +2,10 @@
 
 namespace OmniSharp.Extensions.LanguageServer.Protocol.Document
 {
-    public class TextDocumentAttributes : IEquatable<TextDocumentAttributes>
+    public record TextDocumentAttributes
     {
+        public TextDocumentAttributes() { }
+
         public TextDocumentAttributes(DocumentUri uri, string languageId)
```

Comment:
This doesn't [seem to be the idiomatic way](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#record-types) of assigning properties in a record, though it will probably work as is, so it's mostly a matter of preference (I'm personally loving the typescript-y property assignment/definition.

Reply:
There are some JSON deserialization issues which is why some of these types have default constructors as well, so we might not get all the wins of a records, at least until we can move to System.Text.Json.  SJT is going to be a ways away for now.